lib/SPIRVProducerPass.cpp

```diff
@@ -165,9 +165,8 @@ struct SPIRVOperand {
 
 typedef SmallVector<SPIRVOperand, 4> SPIRVOperandVec;
 
-SPIRVOperandVec &operator<<(SPIRVOperandVec &list, SPIRVOperand elem) {
+void operator<<(SPIRVOperandVec &list, SPIRVOperand elem) {
```

Comment:
I don't follow why this wasn't a problem previously. This change is resulting in a lot of churn in the diff and an unintuitive usage requirement.

Reply:
I agree. The problem is that now constants are added on-the-fly so the getSPIRVValue call has side effect of possibly creating a constant. As well the getSPIRVType can also create a new type, however these are mostly created up front (I would also like to remove the FindType pass but it causes alot of changes in tests).

And so we get to the << operator which can process the inputs in any order (apparently reverse order on Windows). I was considering a new interface where << could take a Type * or a Value *. This way it could preserve the order since the operator overloads would be called in order. I can make that change here and keep the chaining of << but with passing these types instead.

Reply:
So the tests were too fragile and expecting instructions in too specific an order?

Reply:
Yes, that is correct (did not use CHECK-DAG). The generated code was correct in both cases but not consistent across platforms.